src/codecs/native_midi/native_midi_linux_alsa.c

```diff
@@ -464,6 +466,37 @@ NativeMidiSong *native_midi_loadsong_IO(SDL_IOStream *src, bool closeio)
         return NULL;
     }
 
+    /* Since ALSA requires the starting F0 for SysEx, but MIDIEvent.extraData doesn't contain it, we must preprocess the list */
+    /* In addition, since we're going through the list, store the last event's time for looping purposes */
+    do {
+        /* Is this a SysEx? */
+        if (event->status == MIDI_CMD_COMMON_SYSEX && event->extraLen) {
+            /* Sanity check in case something changes in the future */
+            /* This is safe to do since we can't have an F0 manufacturer ID */
+            assert(event->extraData[0] != 0xF0);
```

Comment:
Should we use SDL_assert or one of its variants here? (CC @slouken )

Reply:
For what it's worth, I used assert explicitly so that it gets compiled out in non debug builds.

IIRC SDL_assert makes it into more than just debug, and a normal user is never going to hit. It's only for if there's a change in SDL_mixer itself.